sonic-xcvrd/src/xcvrd_utilities/y_cable_helper.py

```diff
@@ -0,0 +1,510 @@
+    from sonic_y_cable import y_cable
+    from swsscommon import swsscommon
+except ImportError, e:
+    raise ImportError(str(e) + " - required module not found")
+
+
+SELECT_TIMEOUT = 1000
+
+y_cable_platform_sfputil = None
+y_cable_platform_chassis = None
+
+SYSLOG_IDENTIFIER = "y_cable_helper"
+
+helper_logger = logger.Logger(SYSLOG_IDENTIFIER)
+
+
+# SFP status definition, shall be aligned with the definition in get_change_event() of ChassisBase
+SFP_STATUS_REMOVED = '0'
+SFP_STATUS_INSERTED = '1'
+
+# SFP error codes, stored as strings. Can add more as needed.
+SFP_STATUS_ERR_I2C_STUCK = '2'
+SFP_STATUS_ERR_BAD_EEPROM = '3'
+SFP_STATUS_ERR_UNSUPPORTED_CABLE = '4'
+SFP_STATUS_ERR_HIGH_TEMP = '5'
+SFP_STATUS_ERR_BAD_CABLE = '6'
+
+# Store the error codes in a set for convenience
+errors_block_eeprom_reading = {
+    SFP_STATUS_ERR_I2C_STUCK,
+    SFP_STATUS_ERR_BAD_EEPROM,
+    SFP_STATUS_ERR_UNSUPPORTED_CABLE,
+    SFP_STATUS_ERR_HIGH_TEMP,
+    SFP_STATUS_ERR_BAD_CABLE
+}
+
+# Find out the underneath physical port list by logical name
+def logical_port_name_to_physical_port_list(port_name):
+    if port_name.startswith("Ethernet"):
+        if y_cable_platform_sfputil.is_logical_port(port_name):
+            return y_cable_platform_sfputil.get_logical_to_physical(port_name)
+        else:
+            helper_logger.log_error("Invalid port '%s'" % port_name)
+            return None
+    else:
+        return [int(port_name)]
+
+
+def _wrapper_get_presence(physical_port):
+    if y_cable_platform_chassis is not None:
+        try:
+            return y_cable_platform_chassis.get_sfp(physical_port).get_presence()
+        except NotImplementedError:
+            pass
+    return y_cable_platform_sfputil.get_presence(physical_port)
+
+
+# Delete port from Y cable status table
+def delete_port_from_y_cable_table(logical_port_name, y_cable_tbl):
+    y_cable_tbl._del(logical_port_name)
+
+
+def update_tor_active_side(read_side, status, logical_port_name):
+    physical_port_list = logical_port_name_to_physical_port_list(
+        logical_port_name)
+
+    if len(physical_port_list) == 1:
+
+        physical_port = physical_port_list[0]
+        if int(read_side) == 1:
+            if status == "active":
+                y_cable.toggle_mux_to_torA(physical_port)
+                return 1
+            elif status == "standby":
+                y_cable.toggle_mux_to_torB(physical_port)
+                return 2
+        elif int(read_side) == 2:
+            if status == "active":
+                y_cable.toggle_mux_to_torB(physical_port)
+                return 2
+            elif status == "standby":
+                y_cable.toggle_mux_to_torA(physical_port)
+                return 1
+
+        ''' TODO: Now that mux has been issued a command for toggle, check to see if
+            mux has indeed been toggled
+            This check might not be neccessary
+            active_side = y_cable.check_active_linked_tor_side(physical_port)
```

Comment:
Suggest using comments (`#`) here instead of docstring. Docstrings are intended to document how to _use_ the code, whereas comments are indented to document how the code _works_.

On second thought, I think you should remove this and just replace with a comment:

```python
# TODO: Should we confirm that the mux was indeed toggled?
```

Then, once you have made the determination, you can delete the comment and either add the new code or not. Because this line is not the full code which would be necessary to perform the check, it doesn't make sense to leave it here.

Reply:
fixed